contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -290,10 +296,12 @@ contract MixinStakingPoolRewards is
             return;
         }
 
-        // Transfer from RewardVault to this contract.
-        rewardVault.transfer(poolId, address(uint160(address(this))), balance);
-        // Transfer to EthVault.
-        ethVault.depositFor(member, balance);
+        // Decrement the balance of the pool
+        balanceByPoolId[poolId] = balanceByPoolId[poolId].safeSub(balance);
```

Comment:
It looks like `balanceByPoolId` and `reservedWethBalance` must always move together.

A pattern we've used in similar scenarios is to encapsulate this behavior in a single function, then document in `MixinStorage` that these variables should be modified via the functions opposed to directly.